src/vendor/api/v1/stripe/views.py

```diff
@@ -360,3 +361,26 @@ def post(self, request, *args, **kwargs):
             # Subscription Invoice
             return process_stripe_invoice_subscription_payment_succeded(stripe_invoice, site)
 
+
+class StripeInvoiceUpcomingEvent(StripeBaseAPI):
+    def post(self, request, *args, **kwargs):
+        site = get_site_from_request(request)
+        processor = StripeProcessor(site)
+
+        if not self.is_valid_post(site):
+            logger.error("StripeInvoicePaid error: invalid post")
+            return HttpResponse(status=200, content="StripeInvoicePaid error: invalid post")
+
+        if not self.is_incoming_event_correct(self.event, StripeEvents.INVOICE_UPCOMING):
+            logger.error(f"StripeInvoicePaid error: invalid event {self.event}")
+            return HttpResponse(status=200, content=f"StripeInvoicePaid error: invalid event {self.event}")
+
+        stripe_invoice = self.event.data.object
+        customer_profile, stripe_customer = processor.get_customer_profile_and_stripe_customer(stripe_invoice.customer)
+        if not customer_profile or not stripe_customer:
+            logger.error(f"error retrieving customer information for request: {self.event}")
+            return HttpResponse(status=200)
+        
+        logger.info(f"Upcoming Invoice for stripe_customer: {stripe_customer} customer_profile: {customer_profile}")
+
+        stripe_invoice_upcoming.send(sender=self.__class__, customer_profile=customer_profile)
```

Comment:
so this sends a signal to the customer profile on the TC side, which we'll intercept and send an email in response?

Reply:
Yes, right here we are just triggering an event. We need a listener on TC to catch the event